bind.sh

```diff
@@ -133,15 +156,29 @@ fi
 if [[ -n "${cpus+x}" || -n "${mems+x}" ]]; then
   if command -v numactl &> /dev/null; then
       if [[ -n "${cpus+x}" ]]; then
-          set -- --physcpubind "${cpus[$rank]}" "$@"
+          set -- --physcpubind "${cpus[$local_rank]}" "$@"
       fi
       if [[ -n "${mems+x}" ]]; then
-          set -- --membind "${mems[$rank]}" "$@"
+          set -- --membind "${mems[$local_rank]}" "$@"
       fi
       set -- numactl "$@"
   else
       echo "Warning: numactl is not available, cannot bind to cores or memories" 1>&2
   fi
 fi
 
+# arguments may contain the substring %%LEGATE_GLOBAL_RANK%% which needs to be
+# be replaced with the actual computed rank for downstream processes to use
+updated=()
+for arg in "$@"; do
+  updated+=("${arg/\%\%LEGATE_GLOBAL_RANK\%\%/$LEGATE_GLOBAL_RANK}")
+done
+
+set -- "${updated[@]}"
+
+if [ "$debug" == "1" ]; then
+  echo "bind.sh: $@" 1>&2
```

Comment:
I'd suggest doing it like this https://github.com/nv-legate/quickstart/blob/branch-22.12/common.sh#L116-L117, so that spaces are quoted on the output. Also no need to print to stderr for this message.

Reply:
https://github.com/nv-legate/legate.core/pull/471/commits/fa410818d407d9f14d03044109b1dd3791c16376

Reply:
Do we need to be printing this message to stderr (`1>&2`)? I would prefer to use stderr only for errors and warnings, and AFAICT this message is just informational.

Reply:
I guess I am confused by the standard to apply, previously you had wanted the standard help message on stderr as well. Can certainly change in the next PR for adding `--cprofile`

Reply:
The help message is printed right before we exit with a failing exit code, and the output is used as part of the error message whenever the user passes a wrong argument ("argument --foo not recognized, here's what we accept: ..."). So I think it makes sense to print that to stderr. You could make the argument that it should be printed to stdout if the user asks for `--help` explicitly, but I would rather live with this minor inconsistency than add extra cruft.

Conversely, the information message about binding is not pointing to any warning or error, so I'm just going according to the convention we've used elsewhere in the code (e.g. the launcher itself), where we use stdout for verbose informational messages.

I don't know if this convention is best-practice, so I wouldn't disagree if we decided to use stderr for all our informational messaging instead, but then IMHO we should do it consistently across the layers.

Reply:
it's changed in https://github.com/nv-legate/legate.core/pull/475/commits/20703a649a9ce7f4ffbe502fb4f87f09ea860191